### PR TITLE
drivers: ctimer: Add set_top_value support

### PR DESCRIPTION
**Pull request by mmahadevan108**
Implement the set_top_value. This reserves one of the Match channels to set the top value and to reset the counter. Therefore the number of channels available to the user is reduced by 1.

Also update `test_multiple_alarms` test inside `counter_basic_api/src/test_counter.c` to set the top value relative to the current time. This is so that we avoid the case of setting the top value that is prior to the current counter value.

Fixes Issue #44672

### CODE REVIEW

**Comment by dleach02** [comment]
@hakehuang, can you confirm it fixes the problem in your testbed?

**Comment by hakehuang** [comment]
[twister.log](https://github.com/zephyrproject-rtos/zephyr/files/8548881/twister.log)
Test Pass on mimxrt685_evk

File: drivers/counter/counter_mcux_ctimer.c [hunk 1]
```diff
@@ -12,7 +12,8 @@
 #include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
 LOG_MODULE_REGISTER(mcux_ctimer, CONFIG_COUNTER_LOG_LEVEL);
 
-#define NUM_CHANNELS 4
+/* One of the CTimer channels is reserved to implement set_top_value API */
+#define NUM_CHANNELS 3
```
Comment (hakehuang) [inline drivers/counter/counter_mcux_ctimer.c:16]: ctimer_match_t valid value is 0-3. and we use channel 3 to set the top value as match register.
`
typedef enum _ctimer_match
{
    kCTIMER_Match_0 = 0U, /*!< Timer match register 0 */
    kCTIMER_Match_1,      /*!< Timer match register 1 */
    kCTIMER_Match_2,      /*!< Timer match register 2 */
    kCTIMER_Match_3       /*!< Timer match register 3 */
} ctimer_match_t;
`
and we set
' 
struct mcux_lpc_ctimer_data {
	struct mcux_lpc_ctimer_channel_data channels[NUM_CHANNELS];
};
'
maybe we need set a protection in the mcux_lpc_ctimer_set_alarm channel_id

**Review by hakehuang — APPROVED** [review]
some suggestions on channel_id protection as we have a reserved one, if user follow the reference manual and call channel 3, there will be an out of bound access

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**